drivers/sensor/bmm150/bmm150.c

```diff
@@ -0,0 +1,595 @@
+        if (bmm150_set_odr(dev, preset.odr) < 0) {
+                SYS_LOG_DBG("failed to set ODR to %d",
+                            preset.odr);
+                goto err_poweroff;
+        }
+
+        if (i2c_reg_write_byte(data->i2c, config->i2c_slave_addr,
+                               BMM150_REG_REP_XY,
+                               BMM150_REPXY_TO_REGVAL(preset.rep_xy))
+                               < 0) {
+                SYS_LOG_DBG("failed to set REP XY to %d",
+                            preset.rep_xy);
+                goto err_poweroff;
+        }
+
+        if (i2c_reg_write_byte(data->i2c, config->i2c_slave_addr,
+                               BMM150_REG_REP_Z,
+                               BMM150_REPZ_TO_REGVAL(preset.rep_z)) < 0) {
+                SYS_LOG_DBG("failed to set REP Z to %d",
+                            preset.rep_z);
+                goto err_poweroff;
+        }
+
+        if (bmm150_set_power_mode(dev, BMM150_POWER_MODE_NORMAL, 1)
+                                       < 0) {
+                SYS_LOG_DBG("failed to power on device");
+	}
+
+	if (i2c_burst_read(data->i2c, config->i2c_slave_addr,
+                           BMM150_REG_TRIM_START, (u8_t *)&data->tregs,
+                           sizeof(data->tregs)) < 0) {
+                SYS_LOG_DBG("failed to read trim regs");
+                goto err_poweroff;
+        }
+
+        data->rep_xy = 0;
+        data->rep_z = 0;
+        data->odr = 0;
+        data->max_odr = 0;
+        data->sample_x = 0;
+        data->sample_y = 0;
+        data->sample_z = 0;
+
+        data->tregs.xyz1 = sys_le16_to_cpu(data->tregs.xyz1);
+        data->tregs.z1 = sys_le16_to_cpu(data->tregs.z1);
+        data->tregs.z2 = sys_le16_to_cpu(data->tregs.z2);
+        data->tregs.z3 = sys_le16_to_cpu(data->tregs.z3);
+        data->tregs.z4 = sys_le16_to_cpu(data->tregs.z4);
+
+        return 0;
+
+err_poweroff:
+        bmm150_set_power_mode(dev, BMM150_POWER_MODE_NORMAL, 0);
+        bmm150_set_power_mode(dev, BMM150_POWER_MODE_SUSPEND, 1);
+        return -EIO;
+}
+
+static int bmm150_init(struct device *dev)
+{
+        const struct bmm150_config * const config =
+                                          dev->config->config_info;
+        struct bmm150_data *data = dev->driver_data;
+
+        data->i2c = device_get_binding(config->i2c_master_dev_name);
+        if (!data->i2c) {
+                SYS_LOG_DBG("i2c master not found: %s",
+                           config->i2c_master_dev_name);
+                return -EINVAL;
+        }
+
+        if (bmm150_init_chip(dev) < 0) {
+                SYS_LOG_DBG("failed to initialize chip");
+                return -EIO;
+        }
+
+        dev->driver_api = &bmm150_api_funcs;
+
+        return 0;
+}
+
+static const struct bmm150_config bmm150_config = {
+        .i2c_master_dev_name = CONFIG_BMM150_I2C_MASTER_DEV_NAME,
+        .i2c_slave_addr = BMM150_I2C_ADDR,
```

Comment:
You can use CONFIG_BMM150_I2C_ADDR directly and remove the extra BMM150_I2C_ADDR macro